test/validate/service_territory_test.py

```diff
@@ -46,3 +46,22 @@ def test_minmax_rows(
             pv.check_max_rows, expected_rows=expected_rows, margin=0.0, df_name=df_name
         )
     )
+
+
+def test_report_year_discrepency_in_demand_hourly_pa_ferc714(pudl_out_orig):
+    """Test if the vast majority of the years in the two date columns line up."""
+    demand_hourly_pa_ferc714 = pudl_out_orig.demand_hourly_pa_ferc714()
```

Comment:
This will take a while to load ~15M rows from SQLite. If there's another test (or the transform) in which the table is already loaded, we might want to slip it in there rather than reading one of the longest tables multiple times.

Reply:
it took one minute to run for me

```python
(pudl-dev) [pudl] pytest test/validate/service_territory_test.py::test_report_year --live-dbs                 13:28:04  ☁  ferc714_mystery_date_fix ☂ ⚡
================================================================== test session starts ==================================================================
platform darwin -- Python 3.11.6, pytest-7.4.3, pluggy-1.3.0 -- /Users/christinagosnell/miniforge3/envs/pudl-dev/bin/python
cachedir: .pytest_cache
hypothesis profile 'default' -> database=DirectoryBasedExampleDatabase(PosixPath('/Users/christinagosnell/code/pudl/.hypothesis/examples'))
rootdir: /Users/christinagosnell/code/pudl
configfile: pyproject.toml
plugins: hypothesis-6.88.1, anyio-4.0.0, console-scripts-1.4.1, cov-4.1.0, mock-3.12.0, typeguard-4.1.5
collected 1 item

test/validate/service_territory_test.py::test_report_year
-------------------------------------------------------------------- live log setup ---------------------------------------------------------------------
2023-11-07 13:28:13 [    INFO] test.conftest:346 Starting unit tests with output path /Users/christinagosnell/code/pudl_work/output
2023-11-07 13:28:13 [    INFO] catalystcoop.pudl.workspace.setup:175 Skipping existing file at /Users/christinagosnell/code/pudl/src/pudl/package_data/settings/etl_full.yml
2023-11-07 13:28:13 [    INFO] catalystcoop.pudl.workspace.setup:175 Skipping existing file at /Users/christinagosnell/code/pudl/src/pudl/package_data/settings/etl_fast.yml
2023-11-07 13:28:14 [    INFO] alembic.runtime.migration:213 Context impl SQLiteImpl.
2023-11-07 13:28:14 [    INFO] alembic.runtime.migration:216 Will assume non-transactional DDL.
2023-11-07 13:28:14 [    INFO] alembic.runtime.migration:619 Running stamp_revision  -> ec80dd91891a
2023-11-07 13:28:14 [   DEBUG] alembic.runtime.migration:827 new branch insert ec80dd91891a
2023-11-07 13:28:14 [    INFO] alembic.runtime.migration:213 Context impl SQLiteImpl.
add report year validation test
2023-11-07 13:28:14 [    INFO] alembic.runtime.migration:216 Will assume non-transactional DDL.
2023-11-07 13:28:14 [    INFO] alembic.runtime.migration:619 Running stamp_revision  -> ec80dd91891a
2023-11-07 13:28:14 [   DEBUG] alembic.runtime.migration:827 new branch insert ec80dd91891a
2023-11-07 13:28:14 [    INFO] alembic.runtime.migration:213 Context impl SQLiteImpl.
2023-11-07 13:28:14 [    INFO] alembic.runtime.migration:216 Will assume non-transactional DDL.
2023-11-07 13:28:14 [    INFO] alembic.runtime.migration:619 Running stamp_revision  -> ec80dd91891a
2023-11-07 13:28:14 [   DEBUG] alembic.runtime.migration:827 new branch insert ec80dd91891a
2023-11-07 13:28:14 [    INFO] alembic.runtime.migration:213 Context impl SQLiteImpl.
2023-11-07 13:28:14 [    INFO] alembic.runtime.migration:216 Will assume non-transactional DDL.
2023-11-07 13:28:14 [    INFO] alembic.runtime.migration:619 Running stamp_revision  -> ec80dd91891a
2023-11-07 13:28:14 [   DEBUG] alembic.runtime.migration:827 new branch insert ec80dd91891a
2023-11-07 13:28:15 [    INFO] test.conftest:288 setting up the pudl_engine fixture
2023-11-07 13:28:15 [    INFO] alembic.runtime.migration:213 Context impl SQLiteImpl.
2023-11-07 13:28:15 [    INFO] alembic.runtime.migration:216 Will assume non-transactional DDL.
2023-11-07 13:28:15 [    INFO] alembic.runtime.migration:619 Running stamp_revision  -> ec80dd91891a
2023-11-07 13:28:15 [   DEBUG] alembic.runtime.migration:827 new branch insert ec80dd91891a
2023-11-07 13:28:15 [    INFO] alembic.runtime.migration:213 Context impl SQLiteImpl.
2023-11-07 13:28:15 [    INFO] alembic.runtime.migration:216 Will assume non-transactional DDL.
2023-11-07 13:28:15 [    INFO] alembic.runtime.migration:619 Running stamp_revision  -> ec80dd91891a
2023-11-07 13:28:15 [   DEBUG] alembic.runtime.migration:827 new branch insert ec80dd91891a
2023-11-07 13:28:15 [    INFO] alembic.runtime.migration:213 Context impl SQLiteImpl.
2023-11-07 13:28:15 [    INFO] alembic.runtime.migration:216 Will assume non-transactional DDL.
PASSED                                                                                                                                            [100%]

=================================================================== warnings summary ====================================================================
test/validate/service_territory_test.py::test_report_year
  /Users/christinagosnell/miniforge3/envs/pudl-dev/lib/python3.11/site-packages/dagster/_core/storage/sqlalchemy_compat.py:13: RemovedIn20Warning: Deprecated API features detected! These feature(s) are not compatible with SQLAlchemy 2.0. To prevent incompatible upgrades prior to updating applications, ensure requirements files are pinned to "sqlalchemy<2.0". Set environment variable SQLALCHEMY_WARN_20=1 to show all deprecation warnings.  Set environment variable SQLALCHEMY_SILENCE_UBER_WARNING=1 to silence this message. (Background on SQLAlchemy 2.0 at: https://sqlalche.me/e/b8d9)
    return db.select(items)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
======================================================== 1 passed, 1 warning in 83.09s (0:01:23) ========================================================
```

Reply:
i couuuuld add it into `test_minmax_rows` but that feels... wrong.

Reply:
oh hm there is currently no validation test for think chonker of a table. Since we are now loading it i can convert the test to check for both things.